### ISSUE TITLE
closed

### ISSUE THREAD

**Issue by tesla-cat** [freqtrade/freqtrade#4995]
closed

**Comment by xmatthias** [comment]
The documentation is for market orders - and does not have to cover every edge case.

Depending on the stake-size, using market orders may or may not make sense.
The way you're running it is with "simple" market orders - there's nothing wrong with it - but you give up control over the price you receive - so depending on the pair (and stake-size) - you may be pushing price around.

**Comment by tesla-cat** [comment]
i am not sure what you mean

quick question, based on your

> You can adjust your pricing strategy to use the "market" approach. Basically, you reverse pricing sides - so you always (intentionally) try to match the best bid/ask on the other side. The downside is - you'll pay the spread on every trade - so it's similar to market-orders - however you remain in control of the price - so you don't risk eating up the orderbook by using market orders.

should I set: `"order_types": { "buy": "market", "sell": "market" }`
or: `"order_types": { "buy": "limit", "sell": "limit" }`
which one ?

**Comment by xmatthias** [comment]
It's your decision (it's your money).

The first option uses market prices - guaranteed to fill - but you have little idea at what precise price.
the 2nd uses limit orders - which fill at the specified price or better.

**Comment by tesla-cat** [comment]
# the performance of live is completely the opposite of dry-run !

the following is result from live, whereas in dry run it is the complete opposite, how can I get the same performance?

![image](https://user-images.githubusercontent.com/28215638/119234517-e9e0a400-bb60-11eb-9411-ad7d633147aa.png)

**Comment by xmatthias** [comment]
using reversed sides with orderbook will get you similar results (obviously, you'll need to have this on both dry-run and live).

You do pay spread - so your profit calculation will start with a small negative value (usually around 0.4%).
By the looks of it, you're using a very tight stoploss - and try to rely on even smaller profits - which won't work once orderbooks come into play.

**Comment by tesla-cat** [comment]
this strategy achieve 700% profit per day in dry-run mode 

```python
from pandas import DataFrame
from freqtrade.strategy import IStrategy
from datetime import datetime, timedelta

class s1(IStrategy):
    minimal_roi = { "0": 0.001 }
    stoploss = -0.99

    use_custom_stoploss = True
    def custom_stoploss(self, pair: str, trade, current_time: datetime, current_rate: float, current_profit: float, **kwargs) -> float:
        if current_time - timedelta(minutes=1) > trade.open_date_utc:
            return -0.001
        return 1

    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        return dataframe

    def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[:,'buy'] = 1
        return dataframe

    def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[:,'sell'] = 0
        return dataframe
```

**Comment by xmatthias** [comment]
both your minimal ROI and stoploss are 0.1%.
once you account for fees + spread, your orders will sell for loss almost immediately.

Your strategy is to ALWAYS buy - and "hope" you're profitable. That's not a strategy - not one you can realistically expect to get profits from.

if you reverse order sides - dry-run will perform equally bad.

Without reversing sides, dry-run will perform good (because it assumes orders fill immediately)- and real orders will time out at very high rates - or will fill  late - at which point they're potentially losing.

As mentioned above - your values are too tight to get a realistic profit, and the strategy is no strategy - not once markets come into play.

Please do some research on how markets (in particular orderbooks and order filling) works to understand the issues explained here.
